apps/server/src/modules/tool/service/tool-validation.service.spec.ts

```diff
@@ -293,6 +327,30 @@ describe('ToolValidation', () => {
 			});
 		});
 
+		describe('validateDefaultValue', () => {
```

Comment:
you are testing this behavior for the same context (same public function) 2 times, and try to test it a third time. Either treat it as an implementation detail of a blackbox function, and only test the behavior once, or treat this as a seperate unit, extract it, test it in isolation, and just check if its being used when it should.

As it is right now, you are duplicating code multiple times (and worse, one of the duplications is broken)

Reply:
another PR (you just reviewed) splits the validation service into 2. With that the duplication problem should fix itself after the merge